statistical_disclosure_control/cell_key_perturbation/cell_key_perturbation_specification.md

```diff
@@ -0,0 +1,183 @@
+household or other statistical unit). These keys do not change. (A range of
+0-255 is used for census 2021, other datasets may use a range of 0-4095).
+
+* Cell key -- The sum of record keys for a given cell. Modulo 256 is taken so
+the cell keys remain in the range 0-255. (or modulo 4096 to keep a range of
+0-4095).
+
+* Cell value -- The number of records in cell, ie the number of people with
+given characteristics
+
+* pvalue -- perturbation value. The value of noise added to cells, e.g. +1, -1
+
+* pcv -- perturbation cell value. This is an amended cell value needed to
+merge on the ptable
+
+* ptable -- perturbation table. The lookup file containing the pvalues, this
+determines which cell get perturbed.
+
+## 3.0 Introduction  
+
+---
+
+Cell key perturbation adds noise ('perturbation') to cell counts in frequency
+tables. This noise introduces uncertainty in individual values, and thereby
+introduces uncertainty to potential disclosures. The method is specifically
+designed to protect against disclosure by differencing. This method was used
+in ONS Census 2021.
+
+## 4.0 Assumptions  
+
+---
+
+Cell-Key Perturbation works based on the following assumptions:
+
+* The analyst is producing a frequency table that requires cell key
+perturbation protection.
+
+* The analyst has a ptable available to them, relevant to their dataset,
+supplied by the SDC team or data owners.
+
+* The data used contains one row per record (person, household, business, or
+other individual) and one column for each variable/characteristic.
+
+The method is only suitable for producing frequency tables, and cannot be
+applied without a ptable file. It may or may not be possible to apply the
+method to data that does not contain one row per record and one column per
+variable.
+
+## 5.0 Data records  
+
+---
+
+The following field types will be present in the input and output records.
+
+## 5.1 Input records  
+
+---
+
+<!-- markdownlint-disable -->
+ | Variable Definition |Type of Variable| Format of specific variable (if applicable)| Expected range of the values | Meaning of the values| Expected level of aggregation | Comments | 
+ |:---       |:---     |:---     |:---   |:--- |:--- |:--- |
+ | Record key| Integer | Integer | 0-255 |The numbers are random and have no meaning. The 'keys' are used as inputs in perturbation to apply consistency| One per statistical unit (person, household, business) |  The name of the record key column is supplied to the function. Record key variables may be named as record_key, record_key_person etc |
+<!-- markdownlint-enable -->
+
+* The microdata used is expected to contain one row per record (person,
+household, business, or other individual) and one column for each
+variable/characteristic.
+
+* The microdata needs a column containing record key, random numbers between
+0-255.
+
+* The method needs a ptable file, containing the perturbation/noise to be
+added.
+
+* The function call requires a list of variables to be tabulated
```

Comment:
It would be best if we could remove "function call", the method spec should just be about the algorithm and the overview of the method. It doesn't describe the actual code.